src/plugins/ffmemless/plugin.c

```diff
@@ -499,6 +500,11 @@ static int ffm_setup_effects(const NProplist *props, GHashTable *effects)
 		}
 
 		if (ffm.cache_effects) {
+			// custom data is not needed after upload and would point anyway to stack
+			if (custom_data_applied) {
+				ff.u.periodic.custom_data = NULL;
+				ff.u.periodic.custom_len = 0;
```

Comment:
To me it looks like custom data would be applicable only with FF_PERIODIC type and FF_CUSTOM waveform, which makes me think that custom_data_applied variable might not be needed.
Then again, waveform is not always checked and there was some comments about some devices supporting only custom from periodic options. So maybe the way it is now is better after all?

Reply:
Yea, it should be possible to infer if the custom_data can be touched but I thought it's maybe just simpler with the helper variable. Because there are some unions involved need to be careful when to actually touch them.